include/language.php

```diff
@@ -628,7 +546,17 @@ public function get_display_flag() {
 	 * @return string
 	 */
 	public function get_display_flag_url() {
-		return empty( $this->custom_flag_url ) ? $this->flag_url : $this->custom_flag_url;
+		$flag_url = empty( $this->custom_flag_url ) ? $this->flag_url : $this->custom_flag_url;
+
+		/**
+		 * Filters flag URL. The custom one is passed if exists.
+		 *
+		 * @since 3.4
+		 *
+		 * @param string       $flag_url Flag URL.
+		 * @param PLL_Language $language Current language object.
+		 */
+		return apply_filters( 'pll_get_display_flag_url', $flag_url, $this );
```

Comment:
Maybe a way to avoid a new filter :
```php
$url = site_url( set_url_scheme( $url, 'relative' ) );
```
with a comment to explain what we are doing.

Reply:
Good idea! Done in 1ea4829.